src/topology/algebra/group.lean

```diff
@@ -544,8 +562,28 @@ begin
     rwa [mem_preimage, inv_mul_cancel_right] }
 end
 
+
+/-- Every locally compact separable topological group is σ-compact.
+  Note: this is not true if we drop the topological group hypothesis. -/
+@[priority 100] instance separable_locally_compact_group.sigma_compact_space
+  [separable_space G] [locally_compact_space G] : sigma_compact_space G :=
```

Comment:
This is true in more general settings:

* for a `[topological_space α] [locally_compact_space α] [second_countable_topology α]`: I have a proof locally; this is not exactly a generalization of your statement because your lemma doesn't need a countable basis of the `uniformity G`;
* for a `[uniform_space α] [locally_compact_space α]` (I have no formalized proof yet): this is a generalization of your lemma because any topological group is a uniform group.